pkg/event/id.go

```diff
@@ -0,0 +1,42 @@
+package event
+
+import (
+	"encoding/hex"
+	"hash/fnv"
+	"strings"
+	"time"
+
+	apiv1 "github.com/acorn-io/acorn/pkg/apis/api.acorn.io/v1"
+)
+
+// ContentID returns a deterministic ID based on the content of a given event.
+// The returned ID is a valid kubernetes resource name (metadata.name).
+func ContentID(e *apiv1.Event) (string, error) {
+	// TODO: Reduce the field set used to generate when composite events are added.
+	// TODO: Find a better way of selecting and encoding field sets. Maybe a multi-layered io.Writer.
+	fieldSet := strings.Join([]string{
+		e.Type,
+		string(e.Severity),
+		e.Actor,
+		e.Source.String(),
+		e.Description,
+		e.Observed.UTC().Format(time.RFC3339),
+	}, ",")
+
+	h := fnv.New128a()
+	if _, err := h.Write([]byte(fieldSet)); err != nil {
+		return "", err
+	}
+
+	digest := h.Sum([]byte{})
```

Comment:
nit: I have always seen this
```suggestion
	digest := h.Sum(nil)
```